src/fenicsxconcrete/finite_element_problem/linear_elasticity_grf.py

```diff
@@ -0,0 +1,143 @@
+        self.field_function_space = df.fem.FunctionSpace(self.experiment.mesh, ("CG", 1))
+        self.lambda_ = df.fem.Function(self.field_function_space)
+        self.mu = df.fem.Function(self.field_function_space)
+
+        lame1, lame2 = self.get_lames_constants()
+        self.lambda_.vector[:] = lame1
+        self.mu.vector[
+            :
+        ] = lame2  # make this vector as a fenics constant array. Update the lame1 and lame2 in each iteration.
+
+        # define function space ets.
+        self.V = df.fem.VectorFunctionSpace(self.mesh, ("Lagrange", self.p["degree"]))  # 2 for quadratic elements
+        self.V_scalar = df.fem.FunctionSpace(self.mesh, ("Lagrange", self.p["degree"]))
+
+        # Define variational problem
+        self.u_trial = ufl.TrialFunction(self.V)
+        self.v = ufl.TestFunction(self.V)
+
+        # initialize L field, not sure if this is the best way...
+        zero_field = df.fem.Constant(self.mesh, ScalarType(np.zeros(self.p["dim"])))
+        self.L = ufl.dot(zero_field, self.v) * ufl.dx
+
+        # apply external loads
+        external_force = self.experiment.create_force_boundary(self.v)
+        if external_force:
+            self.L = self.L + external_force
+
+        body_force = self.experiment.create_body_force(self.v)
+        if body_force:
+            self.L = self.L + body_force
+
+        # boundary conditions only after function space
+        bcs = self.experiment.create_displacement_boundary(self.V)
+
+        self.a = ufl.inner(self.sigma(self.u_trial), self.epsilon(self.v)) * ufl.dx
+        self.weak_form_problem = df.fem.petsc.LinearProblem(
+            self.a,
+            self.L,
+            bcs=bcs,
+            petsc_options={"ksp_type": "preonly", "pc_type": "lu"},
+        )
+
+    # Random E and nu fields.
+    def random_field_generator(
+        self,
+        field_function_space,
+        cov_name,
+        mean,
+        correlation_length1,
+        correlation_length2,
+        variance,
+        no_eigen_values,
+        ktol,
+    ):
+        random_field = Randomfield(
+            field_function_space,
+            cov_name,
+            mean,
+            correlation_length1,
+            correlation_length2,
+            variance,
+            no_eigen_values,
+            ktol,
+        )
+        # random_field.solve_covariance_EVP()
+        return random_field
+
+    def parameters_conversion(self, lognormal_mean, lognormal_sigma):
+        from math import sqrt
+
+        normal_mean = np.log(lognormal_mean / sqrt(1 + (lognormal_sigma / lognormal_mean) ** 2))
+        normal_sigma = np.log(1 + (lognormal_sigma / lognormal_mean) ** 2)
+        return normal_mean, normal_sigma
+
+    def get_lames_constants(
+        self,
+    ):
+        # Random E and nu fields.
+        E_mean, E_variance = self.parameters_conversion(self.p["E"], 100e9)  # 3
+        Nu_mean, Nu_variance = self.parameters_conversion(self.p["nu"], 0.3)  # 0.03
+
```

Comment:
The sigma values should be possible to pass as parameters. The same for the static values in the random field generation.